core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/datalake/upgrade/ClusterUpgradeService.java

```diff
@@ -32,26 +35,52 @@
     @Inject
     private StackUpdater stackUpdater;
 
-    public void upgradeClusterManager(long stackId, StatedImage targetImage) {
+    public boolean upgradeClusterManager(long stackId, StatedImage currentImage, StatedImage targetImage) {
+        String currentCmBuildNumber = currentImage.getImage().getCmBuildNumber();
+        boolean clusterManagerUpdateNeeded = !currentCmBuildNumber.equals(targetImage.getImage().getCmBuildNumber());
```

Comment:
clusterManagerUpdateNeeded and clusterRuntimeUpgradeNeeded are used twice, could you pls extract those to separate methods?